ApollosProject/apollos-prototype
# Adds Like/Unlike using Following rock table

## DESCRIPTION

### What does this PR do, or why is it needed?

- Removes `interactions` as the Rock backend for `like/unlike`, replaced with Shareable. 
  - Way faster / way less Rock calls required to `Like`/`Unlike`
  - More idiomatic with Rock
- Improves the `rock-interactions` analytics interface.
  - Previously it wasn't always saving data to the table.
  - Now the item name and a deep link to the item will show up in the interactions table.
- Experimental modularization for `Shareable`. Now if you don't include shareable, you won't see errors related to it's schema types being missing! I plan on making similar changes to all interdependent fields in the future. 
- Adds a `MIGRATION.md` doc to describe how to safely update to this version. 
- Changes `RockConstants` so it's functionality is more clear. 
- Created `data-connector-rock-actions` module which contains `Interactions`, `Likeable`, and `RockConstants`
- Maybe some other stuff I'm forgetting.  


### What design trade-offs/decisions were made?
Made the choice to postpone further modularization of the schema into the future (it's not needed at this point)

Made to choice to postpone adding pagination to the `userLikedItems` query (will require front end changes). 

### How do I test this PR?

-Open the app and login
- Like an item
  - heart icon should turn bold
  - item should show up in your I'm Following list in the connect tab
- From the Connect Tab, I'm Following
 - Unlike an Item. It should disappear from the I'm Following List. The heart icon should now longer be bold on the Item screen in the Home Page. 

### What automated tests did you write?
Wrote new tests for Followable, Interactions, and part of Rock Constants. Increased test coverage by 1% :D 

## TODO:

- [x] Closes #348 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
